File: Algorithms/p199_Binary_Tree_Right_Side_View_1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> res;
        if(root){
            deque<TreeNode*> queue;
            queue.push_back(root);
            while(!queue.empty()){
                res.push_back(queue.front()->val);
                int size = queue.size();
                for(int i=0;i<size;++i){
                    if(queue.front()->right) queue.push_back(queue.front()->right);
                    if(queue.front()->left) queue.push_back(queue.front()->left);
                    queue.pop_front();
                }
            }
        }
        return res;
    }
};

File: Algorithms/p199_Binary_Tree_Right_Side_View/p199_Binary_Tree_Right_Side_View_1.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def rightSideView(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        res = []
        if root:
            queue = collections.deque()
            queue.append(root)
            while queue:
                res.append(queue[0].val)
                size = len(queue)
                for i in range(0,size):
                    if queue[0].right:
                        queue.append(queue[0].right)
                    if queue[0].left:
                        queue.append(queue[0].left)
                    queue.popleft()
        return res
